lib/Migration/Version5000Date20211025124248.php

```diff
@@ -0,0 +1,218 @@
+	];
+
+	/** @var IDBConnection */
+	private $dbc;
+
+	public function __construct(IDBConnection $dbc) {
+		$this->dbc = $dbc;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		if (!$schema->hasTable('user_saml_configurations')) {
+			$table = $schema->createTable('user_saml_configurations');
+			$table->addColumn('id', Types::INTEGER, [
+				'notnull' => true,
+			]);
+			$table->addColumn('name', Types::STRING, [
+				'length' => 256,
+				'notnull' => false,
+				'default' => '',
+			]);
+			$table->addColumn('configuration', Types::TEXT, [
+				'notnull' => true,
+			]);
+			$table->setPrimaryKey(['id'], 'idx_user_saml_config');
+		}
+
+		return $schema;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		$prefixes = $this->fetchPrefixes();
+		foreach ($prefixes as $prefix) {
+			$keyStart = $prefix === 1 ? '' : $prefix . '-';
+			$configKeys = array_reduce(
+				self::IDP_CONFIG_KEYS,
+				function (array $carry, string $rawConfigKey) use ($keyStart): array {
+					$carry[] = $keyStart . $rawConfigKey;
+					return $carry;
+				},
+				[]
+			);
+
+			$configData = $keysToDelete = [];
+			$gRows = $this->readConfiguration($configKeys);
+			while ($row = $gRows->current()) {
+				$configData[$this->normalizeKey($row['configkey'])] = $row['configvalue'];
+				$keysToDelete[] = $row['configkey'];
+				$gRows->next();
+			}
+
+			if ($this->insertConfiguration($prefix, $configData) && !empty($keysToDelete)) {
+				$this->deleteOldConfiguration($keysToDelete);
+			}
+		}
+
+		$this->deletePrefixes();
+	}
+
+	protected function deleteOldConfiguration(array $keys): bool {
+		static $deleteQuery;
+		if (!$deleteQuery) {
+			$deleteQuery = $this->dbc->getQueryBuilder();
+
+			$deleteQuery->delete('appconfig')
+				->where($deleteQuery->expr()->eq('appid', $deleteQuery->createNamedParameter('user_saml')))
+				->andWhere($deleteQuery->expr()->in('configkey', $deleteQuery->createParameter('cfgKeys')));
+		}
+
+		$deletedRows = $deleteQuery
+			->setParameter('cfgKeys', $keys, IQueryBuilder::PARAM_STR_ARRAY)
+			->executeStatement();
+
+		return $deletedRows > 0;
+	}
+
+	protected function insertConfiguration(int $id, array $configData): bool {
+		static $insertQuery;
```

Comment:
should this be a class attribute ?

Reply:
Also did that with readQuery and deleteQuery. I also noticed that readQuery wasn't correctly put inside the if statement